res/controllers/common-controller-scripts.js

```diff
@@ -140,6 +140,77 @@ var colorCodeToObject = function(colorCode) {
 var script = function() {
 };
 
+/**
+ * Wraps the engine timer API in JS promises
+ *
+ * Use `new TimerPromise()` to call `engine.beginTimer`, use {@link TimerPromise#cancel} to call `engine.stopTimer()`.
+ * The Promise resolves the first time the timer completes and rejects when canceled.
+ */
+class TimerPromise extends Promise {
```

Comment:
I am not aware of a previous discussion on this matter.

So the idea here is to offer a more modern and ES6+ way of using timers. Promises a particularly suited for this kind of task.

Of course, since Promises only resolve once, it makes more sense for one-shot timers. But the `cancel()` method I added also handle the case where the timer was already cancelled which has the advantage of avoiding the warning when trying to `engine.stopTimer()` a one-shot that was already executed. This warning message really bothers me.

Of course, if this is too controversial, I'll remove this.